samples/nrf9160/gnss/src/main.c

```diff
@@ -231,45 +109,160 @@ static void gnss_event_handler(int event)
+	if (err) {
+		printk("Failed to deactivate LTE, error: %d\n", err);
+
+		return;
+	}
+
+	printk("LTE deactivated\n");
+}
+#endif /* CONFIG_GNSS_SAMPLE_LTE_ON_DEMAND */
+
+static void agps_data_get(struct k_work *item)
+{
+	ARG_UNUSED(item);
+
+	int err;
+
+	requesting_assistance = true;
+
+	printk("\033[1;1H");
+	printk("\033[2J");
+	printk("Assistance data needed, ephe 0x%08x, alm 0x%08x, flags 0x%02x\n",
+		last_agps.sv_mask_ephe,
+		last_agps.sv_mask_alm,
+		last_agps.data_flags);
+
+#if defined(CONFIG_GNSS_SAMPLE_LTE_ON_DEMAND)
+	lte_activate();
+#endif /* CONFIG_GNSS_SAMPLE_LTE_ON_DEMAND */
+
+	err = assistance_request(&last_agps);
+	if (err) {
+		printk("Failed to request assistance data\n");
+	}
+
+#if defined(CONFIG_GNSS_SAMPLE_LTE_ON_DEMAND)
+	lte_deactivate();
+#endif /* CONFIG_GNSS_SAMPLE_LTE_ON_DEMAND */
+
+	requesting_assistance = false;
+}
+#endif /* !CONFIG_GNSS_SAMPLE_ASSISTANCE_NONE */
+
+static int modem_init(void)
+{
+	if (strlen(CONFIG_GNSS_SAMPLE_AT_MAGPIO) > 0) {
+		if (nrf_modem_at_printf("%s", CONFIG_GNSS_SAMPLE_AT_MAGPIO) != 0) {
+			printk("Failed to set MAGPIO configuration\n");
+			return -1;
+		}
+	}
+
+	if (strlen(CONFIG_GNSS_SAMPLE_AT_COEX0) > 0) {
+		if (nrf_modem_at_printf("%s", CONFIG_GNSS_SAMPLE_AT_COEX0) != 0) {
+			printk("Failed to set COEX0 configuration\n");
+			return -1;
+		}
+	}
+
+	if (lte_lc_init() != 0) {
 		printk("Failed to initialize LTE link controller\n");
 		return -1;
 	}
 
-	lte_lc_register_handler(lte_handler);
+#if defined(CONFIG_GNSS_SAMPLE_ASSISTANCE_NONE) || defined(CONFIG_GNSS_SAMPLE_LTE_ON_DEMAND)
+#if defined(CONFIG_GNSS_SAMPLE_LTE_ON_DEMAND)
+	lte_lc_register_handler(lte_lc_event_handler);
+#endif /* CONFIG_GNSS_SAMPLE_LTE_ON_DEMAND */
 
-	static struct supl_api supl_api = {
-		.read       = supl_read,
-		.write      = supl_write,
-		.handler    = inject_agps_type,
-		.logger     = supl_logger,
-		.counter_ms = k_uptime_get
+	if (lte_lc_func_mode_set(LTE_LC_FUNC_MODE_ACTIVATE_GNSS) != 0) {
+		printk("Failed to enable GNSS\n");
+		return -1;
+	}
+#else
+	lte_lc_psm_req(true);
+
+	if (lte_lc_connect() != 0) {
+		printk("Failed to connect to LTE network\n");
+		return -1;
+	}
+
+	printk("Connected to LTE network\n");
+#endif
+
+	return 0;
+}
+
+#if !defined(CONFIG_GNSS_SAMPLE_ASSISTANCE_NONE)
+static int assist_init(void)
+{
+	struct k_work_queue_config cfg = {
+		.name = "agps_work_q",
+		.no_yield = false
 	};
 
 	k_work_queue_start(
 		&agps_work_q,
 		agps_workq_stack_area,
 		K_THREAD_STACK_SIZEOF(agps_workq_stack_area),
 		AGPS_WORKQ_THREAD_PRIORITY,
-		NULL);
+		&cfg);
 
-	k_work_init(&get_agps_data_work, get_agps_data);
+	k_work_init(&agps_data_get_work, agps_data_get);
 
-	if (supl_init(&supl_api) != 0) {
-		printk("Failed to initialize SUPL library\n");
-		return -1;
-	}
-#endif /* CONFIG_SUPL_CLIENT_LIB */
-
-	if (setup_modem() != 0) {
-		printk("Failed to initialize modem\n");
-		return -1;
-	}
+	return assistance_init(&agps_work_q);
```

Comment:
You now have assistance_init() and assist_init() function names. Could you call one of them e.g. pgps_init() or similar as it seems to be mostly about pgps?

Reply:
`assistance.h` is a generic interface that is used with both nRF Cloud and SUPL, so I don't want to change that. I tried to think about a better name for the initialization function in `main.c`, but couldn't come up with any...

What if I change `assist_init()` to `sample_init()`? Although related to assistance, it does initialize the work queue and work item. And actually `sample_init()` could even call `modem_init()` and `gnss_init_and_start()` instead of calling those directly from main.

Reply:
I see...work_init() could be one candidate as well. Well, you pick whatever you think is most describing. No need to change it just for sake of doing something.